Source/WebCore/rendering/RenderLayerBacking.cpp

```diff
@@ -1190,14 +1190,18 @@ bool RenderLayerBacking::updateConfiguration(const RenderLayer* compositingAnces
         // but this is a runtime decision.
         if (element->usesPlatformLayer())
             m_graphicsLayer->setContentsToPlatformLayer(element->platformLayer(), GraphicsLayer::ContentsLayerPurpose::Model);
+#if ENABLE(MODEL_PROCESS)
+        else if (auto contextID = element->layerHostingContextIdentifier(); contextID && element->document().settings().modelProcessEnabled())
```

Comment:
It looks weird that ANY element with a layerHostingContextIdentifier is considered a HostedModel if modelProcessEnabled is true. Even though that's the only use for now, isn't this sort of a violation of the abstraction?

Reply:
This block is under `else if (is<RenderModel>(renderer())) {` so this should be limited to a <model> element already. So within model elements, if it doesn't have a platform layer, and it has a hosting context, we treat it as a HostedModel.

Is there any other circumstances that I'm missing here?

Reply:
Nope, you're right, I misread!